### ISSUE TITLE
Fix S3881: "Implement IDisposable correctly" has inconsistent behaviour in partial classes

### ISSUE THREAD

**Issue by crypto-rsa** [SonarSource/sonar-csharp#760]
### Description

[Rule S3881](https://jira.sonarsource.com/browse/RSPEC-3881) seems to behave inconsistenly in partial classes where the base class implements `IDisposable` (typically `System.Windows.Forms.Form`) and the derived class overrides `IDisposable.Dispose( bool disposing )`.

### Repro steps
Consider this example:

**BaseForm.cs**
```
public class BaseForm : System.Windows.Forms.Form
{
	protected override void Dispose( bool disposing )
	{
		Console.WriteLine( "Disposing" );

		base.Dispose( disposing );
	}
}
```

**DerivedForm.cs**
```
public partial class DerivedForm : BaseForm
{
}
```

**DerivedForm.Designer.cs**
```
partial class DerivedForm
{
	/// <summary>
	/// Required designer variable.
	/// </summary>
	private System.ComponentModel.IContainer components = null;

	/// <summary>
	/// Clean up any resources being used.
	/// </summary>
	/// <param name="disposing">true if managed resources should be disposed; otherwise, false.</param>
	protected override void Dispose( bool disposing )
	{
		if( disposing && (components != null) )
		{
			components.Dispose();
		}
		base.Dispose( disposing );
	}
	...
}
```

### Expected behavior

The code doesn't produce a violation of S3881.

### Actual behavior

A violation of S3881 is reported. However, when the `Dispose( bool disposing )` implementation is moved from `DerivedForm.Designer.cs` to the main `DerivedForm.cs` file, the problem goes away. The behaviour should not depend on the file the `Dispose` method is defined in.
It is particularly annoying because the code for `DerivedForm` is created automatically by the Visual Studio wizard for creating Windows Forms.

### Known workarounds

Move the `Dispose` implementation to the `DerivedForm.cs` file.

### Related information

* SonarC# Version 3.5.0.1803
* Visual Studio Community 2015

**Comment by michalb-sonar** [comment]
Thanks for reporting @crypto-rsa. We were able to reproduce the problem and will address it.
